POSEIDON/src/test/java/uk/ac/ox/oxfish/biology/tuna/EpoBiologyOnlyScenarioAllSpeciesTest.java

```diff
@@ -0,0 +1,117 @@
+package uk.ac.ox.oxfish.biology.tuna;
+
+import com.google.common.collect.ImmutableMap;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import uk.ac.ox.oxfish.biology.Species;
+import uk.ac.ox.oxfish.geography.currents.CurrentPatternMapSupplier;
+import uk.ac.ox.oxfish.model.FishState;
+import uk.ac.ox.oxfish.model.scenario.EpoAbundanceScenario;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static uk.ac.ox.oxfish.utility.csv.CsvParserUtil.recordStream;
+
+/**
+ * Test to see if the simulated biomass is the same as the expected biomass for the first 4 quarters.
+ * Expected biomasses are generated in the preprocessing pipeline as "biomass_test.csv".
+ */
+public class EpoBiologyOnlyScenarioAllSpeciesTest {
+
+    private final Path testInputs = Paths.get(
+        "inputs", "epo_inputs", "tests"
+    );
+
+    private final Path biologyTestFile = testInputs.resolve("biomass_test.csv");
+
+    @Test
+    public void testRunBiologyOnlyScenario() {
+        final EpoAbundanceScenario scenario = new EpoAbundanceScenario();
+
+        final FishState fishState = new FishState();
+        fishState.setScenario(scenario);
+        scenario.getFadMap().setCurrentPatternMapSupplier(CurrentPatternMapSupplier.EMPTY);
+
+        fishState.start();
+
+        System.out.println("SeaTiles: " + fishState.getMap().getAllSeaTiles().size());
+
+        final Species bet = fishState.getSpecies("Bigeye tuna");
+        final Species yft = fishState.getSpecies("Yellowfin tuna");
+        final Species skj = fishState.getSpecies("Skipjack tuna");
+
+
+        System.out.println("breakpoint");
+
+        Map<Integer, Map<String, Double>> expectedBiomass = recordStream(biologyTestFile)
+            .collect(Collectors.groupingBy(
+                record -> record.getInt("day"),
+                Collectors.toMap(record -> record.getString("species"), record -> record.getDouble("biomass"))
+            ));
+
+        Map<Species, double[][]> speciesAbundances =
+            ImmutableMap.of(
+                bet, fishState.getTotalAbundance(bet),
+                yft, fishState.getTotalAbundance(yft),
+                skj, fishState.getTotalAbundance(skj)
+            );
+
+        Map<Species, double[][]> speciesDeaths =
+            ImmutableMap.of(
+                bet, fishState.getTotalAbundance(bet),
+                yft, fishState.getTotalAbundance(yft),
+                skj, fishState.getTotalAbundance(skj)
+            );
+
+        System.out.println("File found, start stepping model");
+
+        do {
+            System.out.println(fishState.getStep());
+            if (expectedBiomass.containsKey(fishState.getStep())) {
+                speciesAbundances.forEach((s, abundance) -> {

```

Comment:
`abundance` is not used anywhere inside the loop, which means that this line could be replace with
```Java
fishState.getBiology().getSpecies().forEach(s -> {
```
which means that `speciesAbundances` isn't needed... which means that `bet`, `yft` and `skj` aren't needed... which is a good thing because we don't want to hard-code species names if we can avoid it... :-)